azurerm/internal/services/network/tests/resource_arm_virtual_hub_connection_test.go

```diff
@@ -0,0 +1,272 @@
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMVirtualHubConnectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMVirtualHubConnection_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualHubConnectionExists(data.ResourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMVirtualHubConnection_requiresImport(data),
+				ExpectError: acceptance.RequiresImportError("azurerm_virtual_hub_connection"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMVirtualHubConnection_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_virtual_hub_connection", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMVirtualHubConnectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMVirtualHubConnection_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualHubConnectionExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMVirtualHubConnectionExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Network.VirtualHubClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Virtual Hub Connection not found: %s", resourceName)
+		}
+
+		virtualHubId := rs.Primary.Attributes["virtual_hub_id"]
+		id, err := networkSvc.ParseVirtualHubID(virtualHubId)
+		if err != nil {
+			return err
+		}
+
+		resourceGroup := id.ResourceGroup
+		hubName := id.Name
+		name := rs.Primary.Attributes["name"]
+		resp, err := client.Get(ctx, resourceGroup, hubName)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on network.VirtualHubClient: %+v", err)
+		}
+
+		if resp.VirtualHubProperties == nil {
+			return fmt.Errorf("VirtualHubProperties was nil!")
+		}
+
+		props := *resp.VirtualHubProperties
+		if props.VirtualNetworkConnections == nil {
+			return fmt.Errorf("props.VirtualNetworkConnections was nil")
+		}
+
+		conns := *props.VirtualNetworkConnections
+
+		found := false
+		for _, conn := range conns {
+			if conn.Name != nil && *conn.Name == name {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return fmt.Errorf("Connection %q was not found", name)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMVirtualHubConnectionDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Network.VirtualHubClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		if rs.Type != "azurerm_virtual_hub_connection" {
+			continue
+		}
+
+		virtualHubId := rs.Primary.Attributes["virtual_hub_id"]
```

Comment:
same here

Reply:
Here is getting virtual_hub_id not virtual hub connection id. so won't fix.

Reply:
as above